app/components/category-list.js

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from "react";
+import {
+  View,
+  ListView,
+  TouchableOpacity,
+  Animated,
+  StyleSheet
+} from "react-native";
+import Interactable from "react-native-interactable";
+import Icon from "react-native-vector-icons/MaterialIcons";
+
+import Text from "./text";
+import { baseStyles } from "../styles";
+
+const styles = StyleSheet.create({
+  categoryList: {},
+  category: {
+    borderColor: "#ccc",
+    borderWidth: 1,
+    padding: 5,
+    paddingLeft: 20,
+    height: 40
+  },
+  categoryHeader: { padding: 5, height: 40 }
+});
+
+class CategoryList extends Component {
+  render() {
+    const { navigation, categories } = this.props;
+
+    return (
+      <View style={styles.categoryList}>
+        {categories.map(category => {
+          return (
+            <CategoryView
+              navigation={navigation}
+              category={category}
+              key={category.name}
+            />
+          );
+        })}
+      </View>
+    );
+  }
+}
+
+class CategoryView extends Component {
+  componentWillMount() {
+    const { category } = this.props;
+
+    const ds = new ListView.DataSource({
+      rowHasChanged: (r1, r2) => r1 !== r2
+    });
+
+    this.setState({
+      expanded: false,
+      minHeight: null,
+      maxHeight: null,
+      animation: new Animated.Value(),
+      categoryName: category.name,
+      categories: ds.cloneWithRows(
+        category.list.map(item => ({
+          id: item.id,
+          name: item.name
+        }))
+      )
+    });
+  }
+
+  _setMinHeight = e => {
+    this.state.animation.setValue(20);
+    if (!this.state.minHeight) {
+      this.state.minHeight = 20;
```

Comment:
Ditto for dimensions. (ESLint whines when `this.state` is manipulated directly, and when I tried adjusting for that with Mapbox GL dimensions on the main screen, bad things happened.)